lab.R

```diff
@@ -100,41 +104,58 @@ daily_diff = function(x, order_var) {
 
 #' *Write a pipe that takes `filter_df` as input, groups the data by county and FIPS code, sorts the dataframe by date, and then converts the cumulative cases and death counts to daily changes using `daily_diff`.  (`date` is the order variable.)  Assign the result to `daily_df`. Hint: `mutate()` can replace the value of existing variables.* 
 #' 
+daily_df <- filtered_df %>% 
+  group_by(county,fips) %>% 
+  arrange(date) %>% 
+  mutate(cases = daily_diff(cases,date),
+         deaths = daily_diff(deaths,date)) %>% 
+  ungroup()
 
 #' 3. *Finally we need to calculate rates per 1 million residents.  Write a pipe that takes `daily_diff` as input, joins it with the `pop` dataframe using appropriate variables, removes any rows with missing FIPS codes, and constructs the variables `cases_per_pop` and `deaths_per_pop`.  When constructing these variables, multiply by `per_pop` to get rates per 1 million residents.  Assign the result to `covid_df`, since this contains the Covid data for our analysis.*  
 #' 
-
+covid_df <- daily_df %>% 
+  left_join(pop,by=c('state','county','fips')) %>% 
+  filter(!is.na(fips)) %>% 
+  mutate(cases_per_pop=cases/population*per_pop,
+         deaths_per_pop=deaths/population*per_pop)
 
 
 #' # Problem 4 #
 #' 1. *To explore these time-series data visually, we'll want to use line plots of cases or deaths over time.  The line group needs the `group` aesthetic to determine which values should be treated as part of a single line.  Uncomment and fill in the blanks to plot cases per 1,000,000 residents over time for each county.*  
 #' 
 
-# ggplot(covid_df, aes(---, ---, group = ---)) +
-#     geom_line()
+ ggplot(covid_df, aes(date, cases_per_pop, group = county)) +
+     geom_line()
 
 #' 2. *Because there are so many counties, the lines are heavily overplotted.  Modify your code from the last problem to facet by county.  Try both `scales = 'fixed'` and `scales = 'free_y'`.* 
 #' 
-
+ ggplot(covid_df, aes(date, cases_per_pop, group = county)) +
+   geom_line()+
+   facet_wrap(~county,scales='fixed')
+ 
+ ggplot(covid_df, aes(date, cases_per_pop, group = county)) +
+   geom_line()+
+   facet_wrap(~county,scales='free_y')
+ 
 #' 3. *The plot indicates that, on a few days, some counties gain or lose thousands of cases per million residents.  What's up this that?  Hints:  `plotly::ggplotly()` to create an interactive version of the most recent plot.  Use `filter()` to narrow down the data to particular counties during short periods of time, and `View()` to peruse the data after filtering.*  
-#' 
+#' A: Based on the provenance, this can occur when a county corrects an error in the number of cases they’ve reported in the past, or when a state moves cases from one county to another.
```

Comment:
This can explain negative values for new cases, but not the large magnitudes.  It's not like hundreds or thousands of people are being transferred between counties, even sporadically.  Take a closer look at which counties have these high-magnitude gains and losses.